core/src/main/java/org/jruby/ir/IRMethod.java

```diff
@@ -138,17 +139,33 @@ public InterpreterContext builtInterpreterContext() {
      *
      * @return appropriate interpretercontext
      */
-    public synchronized InterpreterContext builtInterperterContextForJavaConstructor() {
-        InterpreterContext interpreterContext = builtInterpreterContext();
+    public ExitableInterpreterContext builtInterperterContextForJavaConstructor() {
+        ExitableInterpreterContext interpreterContextForJavaConstructor = this.interpreterContextForJavaConstructor;
+        if (interpreterContextForJavaConstructor == null) {
+            synchronized (this) {
+                interpreterContextForJavaConstructor = this.interpreterContextForJavaConstructor;
+                if (interpreterContextForJavaConstructor == null) {
+                    interpreterContextForJavaConstructor = doBuiltInterperterContextForJavaConstructor();
+                    this.interpreterContextForJavaConstructor = interpreterContextForJavaConstructor;
+                }
+            }
+        }
+        return interpreterContextForJavaConstructor == ExitableInterpreterContext.NULL ? null : interpreterContextForJavaConstructor;
+    }
+
+    private volatile ExitableInterpreterContext interpreterContextForJavaConstructor;
 
-        if (usesSuper()) { // We know at least one super is in here somewhere
+    private synchronized ExitableInterpreterContext doBuiltInterperterContextForJavaConstructor() {
```

Comment:
This is new so we should update this spelling also.

Reply:
:+1: fixed, also renamed to follow the `Impl` suffix convention as there are other private methods that use that here